contentcuration/contentcuration/frontend/channelList/views/Channel/__tests__/catalogFilterBar.spec.js

```diff
@@ -0,0 +1,78 @@
+import { mount } from '@vue/test-utils';
+import store from '../../../store';
+import router from '../../../router';
+import CatalogFilterBar from '../CatalogFilterBar';
+
+const collection = {
+  id: 'test-collection',
+};
+
+const query = {
+  keywords: 'testing',
+  languages: 'en,es',
+  coach: true,
+  collection: 'some-collection',
+};
+
+function makeWrapper() {
+  return mount(CatalogFilterBar, {
+    sync: false,
+    router,
+    store,
+    computed: {
+      collections() {
+        return [collection];
+      },
+    },
+  });
+}
+
+describe('catalogFilterBar', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = makeWrapper();
+  });
+
+  describe('public collections', () => {
+    it('should list collections if no filters are selected', () => {
+      expect(wrapper.find('[data-test="collection"]').exists()).toBe(true);
+    });
+    it('should filter by collection if one on click', () => {
+      wrapper.find('[data-test="collection"]').trigger('click');
+      expect(wrapper.vm.collection).toBe(collection.id);
+    });
+  });
+
+  describe('removing filters', () => {
+    beforeEach(() => {
+      Object.entries(query).forEach(([key, val]) => {
+        wrapper.vm[key] = val;
+      });
+      router.replace({ query });
+    });
+    it('clear all button should remove all filters', () => {
+      wrapper.find('[data-test="clear"]').trigger('click');
+      expect(wrapper.keywords).toBeUndefined();
```

Comment:
Might be good to have some sanity checks that the methods don't affect other query params?